llvm/unittests/ProfileData/MemProfTest.cpp

```diff
@@ -747,6 +752,161 @@ TEST(MemProf, YAMLParser) {
                                ElementsAre(0x3000)))));
 }
 
+static std::string ErrorToString(Error E) {
+  std::string ErrMsg;
+  llvm::raw_string_ostream OS(ErrMsg);
+  llvm::logAllUnhandledErrors(std::move(E), OS);
+  return ErrMsg;
+}
+
+// Test the various scenarios when DataAccessProfData should return error on
+// invalid input.
+TEST(MemProf, DataAccessProfileError) {
```

Comment:
Maybe the DataAccessProf tests should be moved to a dedicated DataAccessProfTest.cpp file to mirror where they are defined?